drivers/SmartThings/jbl/src/jbl/api.lua

```diff
@@ -0,0 +1,165 @@
+local log = require "log"
+local json = require "st.json"
+local RestClient = require "lunchbox.rest"
+local utils = require "st.utils"
+local cosock = require "cosock"
+local socket = require "cosock.socket"
+local ssl = require "cosock.ssl"
+
+local jbl_api = {}
+jbl_api.__index = jbl_api
+
+local CREDENTIAL_TIME_OUT_SECONDS = 30
+
+local SSL_CONFIG = {
+    mode = "client",
+    protocol = "any",
+    verify = "peer",
+    options = "all",
+    cafile="./selfSignedRoot.crt"
+}
+
+local ADDITIONAL_HEADERS = {
+    ["Accept"] = "application/json",
+    ["Content-Type"] = "application/json",
+}
+
+function jbl_api.socket_builder(host, port, wrap_ssl)
```

Comment:
Similar to the comment on in the Event Source, it's strongly encouraged that a timeout be set on any socket that can end up in `receive` position for select to avoid yielding a coroutine and never resuming.

Reply:
change to

> function jbl_api.labeled_socket_builder(label)
    local socket_builder = utils.labeled_socket_builder(label, SSL_CONFIG)
    return socket_builder
end